### DISCUSSION TITLE
Channel Backup

### DISCUSSION THREAD

**Discussion by dubem2** [Ride-The-Lightning/RTL#1275] [LND]
Hello all, I am having issues backing up my lnd channels on RTL. I have specified the directory in the RTL-Config.json and still get this error when i try to backup my channels on the UI

<img width="1403" alt="Screen Shot 2023-08-14 at 12 27 51 PM" src="https://github.com/Ride-The-Lightning/RTL/assets/60652826/e44e1b38-cdbb-49d6-b6f8-01708e33f017">

</br>
Inputs will be really appreciated

**Comment by ShahanaFarooqui** [comment]
Can you please check RTL server logs? Above screen doesn't show any reason for the failure. But usually backup fails due to user permission issues.

**Reply by dubem2** [reply]
Here is a screenshot of the server logs

<img width="1422" alt="Screen Shot 2023-08-14 at 10 51 04 PM" src="https://github.com/Ride-The-Lightning/RTL/assets/60652826/709121f4-e9a3-4b3b-b285-22fb02105ed4">
</br>
</br>
I made the backup directory writable with "sudo chmod -R a+rwx /home/ubuntu/RTL/lndchannelbackup". Is there something i am missing?

**Reply by ShahanaFarooqui** [reply]
Your setup looks correct but `ENOENT` is either thrown because the directory does not exist or doesn't have the right permissions. You can confirm your directory permissions with `ls -la /home/ubuntu/RTL` command. Also check if you have any existing files/folders in lndchannelbackup folder with  `ls -la /home/ubuntu/RTL/lndchannelbackup` and your `channelBackupPath` configuration.

**Reply by dubem2** [reply]
directory permissions are set as seen below:

<img width="774" alt="Screen Shot 2023-08-15 at 12 02 10 AM" src="https://github.com/Ride-The-Lightning/RTL/assets/60652826/13e19779-7ea4-4085-a943-0c0446621c77">
</br>
</br>
ls -la /home/ubuntu/RTL/lndchannelbackup shows:
</br>
<img width="862" alt="Screen Shot 2023-08-15 at 12 06 08 AM" src="https://github.com/Ride-The-Lightning/RTL/assets/60652826/4b3399a4-d9b4-473a-b6a6-38dbbcba612d">
</br>
</br>
and finally the config path is set in RTL-Config.json thus:

<img width="862" alt="Screen Shot 2023-08-15 at 12 14 35 AM" src="https://github.com/Ride-The-Lightning/RTL/assets/60652826/c8ea1f14-5e5c-4208-a9e8-aa94b712046e">

**Reply by ShahanaFarooqui** [reply]
I followed all above mentioned steps and still able to download "all" and "channel" backups. So it is definitely related to your system permissions but cannot pinpoint the issue without access.

I was unable to see `channelBackupPath` configuration in the screenshot but make sure that it is set as: 
```
      "Settings": {
        ...
        "channelBackupPath": "lndchannelbackup"
       ...
      }

```

You can also try deleting/moving your old channel-all.bak file as it might have been corrupted for some unknown reason. 

I would also recommend to remove `executable` permissions from this directory & file. Though it is not an issue for now, it will be in future.

**Reply by dubem2** [reply]
I have followed all you stated above but still unable to backup or download backups. below is where i specified the channelBackupPath:
<img width="714" alt="Screen Shot 2023-08-15 at 9 55 08 AM" src="https://github.com/Ride-The-Lightning/RTL/assets/60652826/26471db0-64c4-4197-b00e-8577005807bd">
